README.md

```diff
@@ -239,6 +239,21 @@ node 'sensu-client.foo.com' {
    }
 }
 ```
+
+### Sensu client with Sensu Go module
+
+The following is an example of using the [Sensu Go Puppet module](https://github.com/sensu/sensu-puppet) on the same host using this sensuclassic module.
+
+```puppet
+include sensu::agent
+class { 'sensuclassic':
+  manage_user => false,
+  ssl_dir     => '/etc/sensu/ssl-classic',
+}
+```
+
+**NOTE**: It's important that the inclusion of `sensuclassic` module come **AFTER** the inclusion of `sensu::agent`.
```

Comment:
Could they use a metaparameter such as

```puppet
  require => Class['sensu::agent'],
```

Reply:
That would not help, that would only set resource order but resource order is not the issue. The issue is evaluation order not resource order so you have to ensure Puppet parses the manifest code for `sensu` module before it reaches `sensuclassic` and the use of `defines()` which is evaluation order dependent.  It's sort of like not being able to reference a variable like `$class::foo` before `include class`, there are certain things in Puppet that have to be ordered correctly for evaluation.

Reply:
If the classic module is not referencing the go module, why does this matter?

Reply:
Because we do this:

```
if ! defined(Yumrepo['sensu']) {
  yumrepo { 'sensu': ensure => 'absent' }
}
```

If you include sensuclassic before sensu module then you will get duplicate resources because `defined()`would have already been evaluated when sensu module is parsed and evaluated so you end up with duplicate `Yumrepo['sensu']` resources.  The evaluation order matters when dealing with one resource that is wrapped by `if defined()` and one resource that is not.

Because we are renaming a resource it would be very poor practice to no cleanup the past resource but we also want to ensure this module will not cleanup the resource added if someone uses Sensu Go module at the same time so we have to add support for conditionally removing the old yum repo if not already defined in Puppet.

This order of evaluation issue was already encountered with sensu module which is why we always document and must load the `sensu` class before `sensu::agent`.  I don't recall what in that module had that issue but we've seen this before, some situations in Puppet are evaluation order specific and evaluation order can sometimes matter.

Reply:
Figured example might help.

The error:

```
[root@centos-7 /]# cat /tmp/test.pp 
class { 'sensuclassic':
manage_user => false,
}
include sensu::agent

[root@centos-7 /]# puppet apply /tmp/test.pp 
Warning: Facter: timeout option is not supported for custom facts and will be ignored.
Error: Evaluation Error: Error while evaluating a Resource Statement, Duplicate declaration: Yumrepo[sensu] is already declared at (file: /etc/puppetlabs/code/modules/sensuclassic/manifests/repo/yum.pp, line: 41); cannot redeclare (file: /etc/puppetlabs/code/environments/production/modules/sensu/manifests/repo.pp, line: 16) (file: /etc/puppetlabs/code/environments/production/modules/sensu/manifests/repo.pp, line: 16, column: 5) on node centos-7
```

The working example:

```
[root@centos-7 /]# cat /tmp/test.pp 
include sensu::agent
class { 'sensuclassic':
manage_user => false,
}

[root@centos-7 /]# puppet apply /tmp/test.pp 
Warning: Facter: timeout option is not supported for custom facts and will be ignored.
Notice: Compiled catalog for centos-7 in environment production in 0.47 seconds
Notice: /Stage[main]/Sensu::Common/User[sensu]/home: home changed '/opt/sensu' to '/var/lib/sensu'
Notice: /Stage[main]/Sensu::Repo/Yumrepo[sensu]/ensure: created
<SNIP rest of succesful run>
```

Reply:
I remembered why this also happens in other places and it's evaluation order related:

No errors:

```
class { 'sensu': }
include sensu
```

Errors:

```
include sensu
class { 'sensu': }
```

While not directly related to issue with `defined()`, it's another example of evaluation order in Puppet and the issue we've seen in the past with having to document how to order things to keep from getting errors.

Reply:
Instead of introducing the order dependence, let's just remove the "if not defined, ensure the repo is absent" part of the code and instead document the change of repo name and suggest that if users are not purging their repo directories that they should add an ensure absent to the profile code they are using.

Reply:
I removed the `defined()` conditions and added section under upgrade nodes of README about how to remove the previous repos.  I made the assumption the next version would be v3.8.0.